tools/gen_asdl/asdl_js.py

```diff
@@ -127,13 +127,16 @@ def emit(s, depth=depth):
             self.emit(s, depth)
 
         emit(f"/* ----- {name} ----- */")
-        emit(f"export class {name} extends AST {{}}")
+        emit(f"export class {name} extends AST {{kind=0;}}")
         self.emit_tp_name(name)
         emit("")
+        for i in range(len(sum.types)):
+            type = sum.types[i]
+            emit(f"export const {type.name}_kind = {i+1};")
```

Comment:
we've also got some of these `kind`'s in python but they're not generated

Reply:
Yeah I i think they're somewhere in skulpt's ast file just copied and pasted somewhere. 

In skulpt we use typically use `.constructor` for switch case statements. 

But. Typescript has a known bug for the .constructor property of an object. (It thinks .constructor is always Object 🤦‍♂️) and it would mean faffing about with the any type anytime you wanted to use .constructor. So I figured why not use the same case switch style as cpython while we're here. And then since cpython generates these in Python-asdl.h file we might as well too.

Reply:
referencing the .constructor switch statement issue in typescript https://github.com/microsoft/TypeScript/issues/3841

Reply:
other notes on this. `kind` seems to be completely internal for cpython. And as far as I can tell we don't use these in skulpt. So perhaps we should just remove these and use the skulpt constructor switch case see 89bd2b3 for a typed version. In fact let's go with the constructor version for now (i'll revert this) and if we want it back in we have this as a reference.

Reply:
see resolution in 294b064 (revert kind) and 65cc82c (include typeof unions for switch constructor statements)

Reply:
Oh cool